data/json/effects_on_condition/melee_eocs.json

```diff
@@ -0,0 +1,1266 @@
+[
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_MELEE_PROF_ATTACK_MONSTER",
+    "//": "If the player attacks a monster, and hits, run EOC to check what weapon type they're using.",
+    "global": false,
+    "eoc_type": "EVENT",
+    "required_event": "character_melee_attacks_monster",
+    "condition": { "math": [ "_hits", "==", "1" ] },
+    "effect": [ { "run_eocs": "EOC_MELEE_PROF_WHAT_CATEGORY" } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_MELEE_PROF_ATTACK_CHARACTER",
+    "//": "If the player attacks a character, run EOC to check what weapon type they're using.",
+    "global": false,
+    "eoc_type": "EVENT",
+    "required_event": "character_melee_attacks_character",
+    "condition": { "math": [ "_hits", "==", "1" ] },
+    "effect": [ { "run_eocs": "EOC_MELEE_PROF_WHAT_CATEGORY" } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_MELEE_PROF_WHAT_CATEGORY",
+    "//": "Checks what weapon categories are held, increments experience, and defines lowest proficiency level of held weapon.",
+    "global": false,
+    "effect": [
+      { "math": [ "u_prof_lowest", "=", "4" ] },
+      {
+        "if": { "u_has_wielded_with_weapon_category": "AUTOMATIC_RIFLES" },
+        "then": [
+          {
+            "if": { "math": [ "u_val('proficiency', 'proficiency_id: prof_auto_rifles_familiar', 'format: percent')", ">=", "100" ] },
+            "then": [
+              {
+                "if": { "math": [ "u_val('proficiency', 'proficiency_id: prof_auto_rifles_pro', 'format: percent')", ">=", "100" ] },
+                "then": [
+                  {
+                    "if": { "math": [ "u_val('proficiency', 'proficiency_id: prof_auto_rifles_master', 'format: percent')", ">=", "100" ] },
+                    "then": [
+                      { "math": [ "u_prof_level", "=", "3" ] },
+                      {
+                        "if": { "math": [ "u_prof_level", "<=", "u_prof_lowest" ] },
+                        "then": { "math": [ "u_prof_lowest", "=", "u_prof_level" ] }
+                      }
```

Comment:
```suggestion
                      { "math": [ "u_prof_lowest", "=", "min(u_prof_level, u_prof_lowest)" ] },
```

Reply:
What is this meant to do?

Reply:
Exactly the same thing as the `if` structure you have, but more concise. Take a look at the documentation for min() and other functions https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/NPCs.md#math-functions

Reply:
Love it! Thanks!